run.sh

```diff
@@ -3,327 +3,247 @@
+    data_path=$(find /data -name '*.sql' -print -quit 2>/dev/null)
+    if [[ ! "$data_path" ]]; then
+        wp --color core install |& logger
+        return
+    fi
 
-  # No backups found
-  if [[ "$( find /data -name "*.sql" 2>/dev/null | wc -l )" -eq 0 ]]; then
-    _wp core install
-    _log_last_exit_colorize "Success: core install" "Error: core install failed!"
+    wp --color db import "$data_path" |& logger
 
-    return
-  fi
+    if [[ -n "$URL_REPLACE" ]]; then
+        wp --color search-replace --skip-columns=guid "$BEFORE_URL" "$AFTER_URL" \
+            | grep --color 'replacement' \
+            |& logger
+    fi
+}
 
-  data_path=$( find /data -name "*.sql" -print -quit )
-  _wp db import "$data_path"
-  _log_last_exit_colorize "Success: db import" "Error: db import failed!"
+# Install / remove plugins based on $PLUGINS in parallel threads
+check_plugins() {
+    (
+        declare -a add_list
+        mapfile -t add_list < <(comm -23 \
+            <(echo "${!plugin_deps[@]}" | tr ' ' '\n' | sort) \
+            <(wp plugin list --field=name | sort))
+
+        if [[ "${#add_list[@]}" -gt 0 ]]; then
+            wp --color plugin install --activate "${add_list[@]}" |& logger
+        fi
+    ) &
+
+    (
+        declare -a remove_list
+        mapfile -t remove_list < <(comm -13 \
+            <(echo "${!plugin_deps[@]}" | tr ' ' '\n' | sort) \
+            <(wp plugin list --field=name | sort))
+
+        if [[ ${#remove_list[@]} -gt 0 ]]; then
+            wp --color plugin uninstall --deactivate "${remove_list[@]}" |& logger
+        fi
+    ) &
+
+    wait
+}
 
-  if [[ -n "$URL_REPLACE" ]]; then
-    wp search-replace --skip-columns=guid "$BEFORE_URL" "$AFTER_URL" --allow-root \
-    | grep 'replacement' \
-    |& _colorize
-  fi
+# Install / remove themes based on $THEMES in parallel threads
+check_themes() {
+    (
+        declare -a add_list
+        mapfile -t add_list < <(comm -23 \
+            <(echo "${!theme_deps[@]}" | tr ' ' '\n' | sort) \
+            <(wp theme list --field=name | sort))
+
+        if [[ "${#add_list[@]}" -gt 0 ]]; then
+            wp --color theme install "${add_list[@]}" |& logger
+        fi
+    ) &
+
+    (
+        declare -a remove_list
+        mapfile -t remove_list < <(comm -13 \
+            <(echo "${!theme_deps[@]}" | tr ' ' '\n' | sort) \
+            <(wp theme list --field=name | sort))
+
+        if [[ ${#remove_list[@]} -gt 0 ]]; then
+            wp --color theme delete "${remove_list[@]}" |& logger
+        fi
+    ) &
+
+    wait
 }
 
-check_plugins() {
-  local key
-  local plugin
-  local to_install=()
-  local to_remove=()
-
-  if [[ "${#plugin_deps[@]}" -gt 0 ]]; then
-    for key in "${!plugin_deps[@]}"; do
-      if ! wp plugin is-installed --allow-root "$key"; then
-        to_install+=( "${plugin_deps[$key]}" )
-      fi
-    done
-  fi
+configure() {
+    # Ensures that this only runs on the initial build
+    [[ -f ~/.wp-cli/config.yml ]] && return
```

Comment:
I'm not sure this take any time so I would run it everytime. It allows that changes to variables are actually passed down to the image. wdyt?

Reply:
I see what you mean. 

The issue currently is that it would append the `ServerName` directive and append the completions line in the bashrc on every restart.

So the end result of that would be a single `ServerName`, but a bunch of `ServerAlias`es and a bunch of `. /etc/bash_completion.d/wp-cli`s.

I can prob adjust the logic there though if you think we should keep the ability to change variable defaults.